nuxeo-features/nuxeo-platform-audit/nuxeo-platform-audit-core/src/main/java/org/nuxeo/ecm/platform/audit/listener/StreamAuditEventListener.java

```diff
@@ -127,7 +128,16 @@ protected void writeEntry(LogAppender<Record> appender, LogEntry entry) {
         if (json == null) {
             return;
         }
-        appender.append(0, Record.of(String.valueOf(entry.getId()), json.getBytes(UTF_8)));
+        long timestamp = getTimestampForEntry(entry);
+        appender.append(0, new Record(String.valueOf(entry.getId()), json.getBytes(UTF_8),
+                Watermark.ofTimestamp(timestamp).getValue()));
+    }
+
+    protected long getTimestampForEntry(LogEntry entry) {
+        if (entry.getEventDate() != null) {
```

Comment:
Just an observation. You chose eventDate rather than logDate for the watermark. What are the implications of that?

Reply:
If you are using eventdate/watermark to compute lag then you probably want to use logDate instead.

Reply:
We are in a sync event listener, the event is not yet written to the audit backend so there is no logDate at this moment, the eventDate looks a good candidate for record watermark, more precise than a current time which will be the same for all the events of the transaction

Reply:
Ok. Makes sense.